mm2src/lp_ordermatch.rs

```diff
@@ -925,21 +924,22 @@ fn maker_order_created_p2p_notify(
     };
 
     let to_broadcast = new_protocol::OrdermatchMessage::MakerOrderCreated(message.clone());
-    let (secret, public) = match &order.p2p_privkey {
+    let (secret, public, peer_id) = match &order.p2p_privkey {
         Some(priv_key) => {
             let key_pair = key_pair_from_secret(&priv_key.0).expect("valid priv key");
-            (priv_key.0, *key_pair.public())
+            let peer_id = peer_id_from_secp_public(key_pair.public_slice()).expect("valid public key");
```

Comment:
Could you please explain this in more details? I think I don't understand how this is related to `Taker->Maker` order conversion.

Reply:
I think it can happen without `Taker->Maker` conversion, it's just one specific case I checked.
I meant that if `TakerOrder::key_pair` or `MakerOrder::key_pair` are invalid (user changes the value in the storage or we change format of the field), then mm2 may panic on parsing this field

Reply:
@sergeyboyko0791 I have removed `expect` on ordermatching messages broadcast, but there are other places in swaps code, where it's unclear how to deal with such invalid state.  Swap simply can't go on if p2p_privkey and/or HTLC privkeys are invalid. But `apply_event` doesn't return anything. E.g.
https://github.com/KomodoPlatform/atomicDEX-API/blob/f417ae6b9035e087761635ba00e44ba526b2c345/mm2src/lp_swap/taker_swap.rs#L615-L621
I guess I will try to figure it out on the next iteration.

Reply:
That's a good idea 🙂 I have started implementing this.